test/chemistry/test_fermionic_transformation.py

```diff
@@ -0,0 +1,178 @@
+            transformation=TransformationType.FULL,
+            qubit_mapping=QubitMappingType.PARITY,
+            two_qubit_reduction=False,
+            freeze_core=False,
+            orbital_reduction=[])
+
+        qubit_op, _ = fermionic_transformation.transform(self.driver)
+        self._validate_vars(fermionic_transformation)
+        self._validate_info(fermionic_transformation)
+        self._validate_input_object(qubit_op)
+
+    def test_bravyi_kitaev(self):
+        """ bravyi kitaev test """
+        fermionic_transformation = FermionicTransformation(
+            transformation=TransformationType.FULL,
+            qubit_mapping=QubitMappingType.BRAVYI_KITAEV,
+            two_qubit_reduction=False,
+            freeze_core=False,
+            orbital_reduction=[])
+
+        qubit_op, _ = fermionic_transformation.transform(self.driver)
+        self._validate_vars(fermionic_transformation)
+        self._validate_info(fermionic_transformation)
+        self._validate_input_object(qubit_op)
+
+    def test_particle_hole(self):
+        """ particle hole test """
+        fermionic_transformation = FermionicTransformation(
+            transformation=TransformationType.PARTICLE_HOLE,
+            qubit_mapping=QubitMappingType.JORDAN_WIGNER,
+            two_qubit_reduction=False,
+            freeze_core=False,
+            orbital_reduction=[])
+
+        qubit_op, _ = fermionic_transformation.transform(self.driver)
+        self._validate_vars(fermionic_transformation, ph_energy_shift=-1.83696799)
+        self._validate_info(fermionic_transformation)
+        self._validate_input_object(qubit_op)
+
+    def test_freeze_core(self):
+        """ freeze core test -- Should be in effect a no-op for H2 """
+        fermionic_transformation = FermionicTransformation(
+            transformation=TransformationType.FULL,
+            qubit_mapping=QubitMappingType.JORDAN_WIGNER,
+            two_qubit_reduction=False,
+            freeze_core=True,
+            orbital_reduction=[])
+
+        qubit_op, _ = fermionic_transformation.transform(self.driver)
+        self._validate_vars(fermionic_transformation)
+        self._validate_info(fermionic_transformation)
+        self._validate_input_object(qubit_op)
+
+    def test_orbital_reduction(self):
+        """ orbital reduction test --- Remove virtual orbital just
+            for test purposes (not sensible!)
+        """
+        fermionic_transformation = FermionicTransformation(
+            transformation=TransformationType.FULL,
+            qubit_mapping=QubitMappingType.JORDAN_WIGNER,
+            two_qubit_reduction=False,
+            freeze_core=False,
+            orbital_reduction=[-1])
+
+        qubit_op, _ = fermionic_transformation.transform(self.driver)
+        self._validate_vars(fermionic_transformation)
+        self._validate_info(fermionic_transformation, num_orbitals=2)
+        self._validate_input_object(qubit_op, num_qubits=2, num_paulis=4)
+
```

Comment:
Since extra aux_ops are now part of the public interface it would be good if that aspect were tested here too

Reply:
Added in 850c79975